### PR TITLE
Topology support

### LINKED ISSUE #48: Add support for Topology
CSI spec v1.0 supports topology. This needs to be incorporated into the operator and ensure it works with X_CSI_TOPOLOGIES and X_CSI_NODE_TOPOLOGY in Ember-CSI driver.

### PR DESCRIPTION
**Pull request by kirankt**
Add csi-topology example directory with an RBD-based example yaml files
Fix bug in CR update during daemonset index 0 creation
Rework topologies yaml to use TopologySelectorLabelRequirement and NodeSelectorRequirement to describe topology and nodes
Re-ran operator-sdk generate k8s to re-generate deepcopy files
Change default topology to deployment specific, unique one
Make glog output more legible in a few places
Removed watching volumesnapshotclass because of error. TODO.
Remove unused nodeselector type. We're using corev1.nodeselector instead
Merge branch 'topology' of https://github.com/kirankt/ember-csi-operator into topology
Closes #48

### CODE REVIEW

File: deploy/examples/csi-topology/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Install a Ceph demo container
+kubectl create -f 00-ceph-demo.yml
```
Comment (cschwede) [inline deploy/examples/csi-topology/deploy.sh]: Suffix needs to be .yaml, not .yml (also for all following files)

File: deploy/examples/csi-topology/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+tar cf system-files.tar etc/ceph/ceph.conf etc/ceph/ceph.client.admin.keyring
+kubectl create -n ember-csi secret generic system-files --from-file=system-files.tar
+[ -e system-files.tar ] && rm -f system-files.tar
+[ -e etc ] && rm -rf etc
+
+# Deploy the operator
+kubectl -n ember-csi create -f 02-operator.yml
+
+echo "Wait until the Operator is ready..."
+kubectl wait -n ceph-demo --timeout=300s --for=condition=Ready pod/ember-csi-operator
```
Comment (cschwede) [inline deploy/examples/csi-topology/deploy.sh]: Needs to use the ember-csi namespace, not ceph-demo

File: deploy/examples/csi-topology/02-operator.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      name: ember-csi-operator
+  template:
+    metadata:
+      labels:
+        name: ember-csi-operator
+    spec:
+      serviceAccountName: ember-csi-operator
+      containers:
+        - name: ember-csi-operator
+          image: quay.io/kirankt/ember-csi-operator:v0.0.5
```
Comment (cschwede) [inline deploy/examples/csi-topology/02-operator.yaml]: There seems to be an issue with this version of the operator, it panics - please see attached logs:
[topoerrors.txt](https://github.com/embercsi/ember-csi-operator/files/3112728/topoerrors.txt)

File: deploy/examples/csi-topology/deploy.sh (hunk 1 above)
Comment (kirankt) [reply deploy/examples/csi-topology/deploy.sh]: The latest commit makes consistent usage of 'yaml' thorough out the code

File: deploy/examples/csi-topology/deploy.sh (hunk 2 above)
Comment (kirankt) [reply deploy/examples/csi-topology/deploy.sh]: Good catch. Fixed in the latest commit.

File: deploy/examples/csi-topology/02-operator.yaml (hunk 3 above)
Comment (kirankt) [reply deploy/examples/csi-topology/02-operator.yaml]: The latest commit fixes this bug. Thanks for the thorough review.

**Comment by cschwede** [comment]
So far my test cases are working, merging this one.

**Pull request merged**